Address issue #845; Add test to ensure regression does not occur

selimnairb opened the pull request:
@pkdash Can you review this PR?  Thanks.

hydrobuild commented:
Can one of the admins verify this patch?

selimnairb commented on hs_core/views/resource_rest_api.py:
```diff
@@ -156,6 +156,8 @@ def get_queryset(self):
         if len(filter_parms['type']) == 0:
             filter_parms['type'] = None
+        else:
+            filter_parms['type'] = list(filter_parms['type'])
```
This is required because hydroshare.get_resource_list can't handle multiple-value filter parameters that are sets (which the serializer returns for this field), but instead requires that lists be used.

pkdash commented:
@selimnairb Looks like 2 of the unit tests are failing.

selimnairb commented:
@pkdash I have fixed the broken new unit tests (which only fail when not run explicitly, which was why I missed it; I was running only the new tests I created, not all tests).

pkdash commented on hs_core/tests/api/rest/test_resource_list.py:
```diff
@@ -0,0 +1,78 @@
+        self.assertEqual(content['results'][1]['resource_id'], geo_pid)
+
+        # Filter by type
+        response = self.client.get('/hsapi/resourceList/', {'type': 'RasterResource'}, format='json')
```
Wondering if there is a need to have a test for passing multiple resource types when calling resourceList api?

selimnairb commented:
@pkdash Thanks for pointing out a limitation in the test (testing only filtering by a single type).  I have updated the test to query by multiple types, which revealed the fact that the user.get_resource_list() did not support querying by multiple types (now fixed).

selimnairb commented:
@pkdash @hyi Please let me know if you see any outstanding problems with this PR, and whether this can be merged.  Thanks.

pkdash commented:
+1
